src/BloomTests/Book/PageMigrationTests.cs

```diff
@@ -0,0 +1,192 @@
+			   <div class='marginBox'>
+					<div class='bloom-imageContainer bloom-leadingElement'><img data-license='cc-by-nc-sa' data-copyright='Copyright © 2012, LASI' style='width: 608px; height: 471px; margin-left: 199px; margin-top: 0px;' src='erjwx3bl.q3c.png' alt='This picture, erjwx3bl.q3c.png, is missing or was loading too slowly.' height='471' width='608'></img></div>
+					<div aria-describedby='qtip-1' data-hasqtip='true' class='bloom-translationGroup bloom-trailingElement normal-style'>
+						<div aria-describedby='qtip-0' data-hasqtip='true' class='bloom-editable normal-style bloom-content1' contenteditable='true' lang='en'>
+							There was an old man called Bilanga who was very tall and also not yet married.
+						</div>
+					</div>
+				</div>
+			</div>
+			");
+			var book = CreateBook();
+			var dom = book.RawDom;
+			var page = (XmlElement) dom.SafeSelectNodes("//div[@id='thePage']")[0];
+			var oldContent = page.OuterXml;
+			book.BringPageUpToDate(page);
+
+			var newPage = (XmlElement) dom.SafeSelectNodes("//div[@id='thePage']")[0];
+			Assert.That(newPage.OuterXml, Is.EqualTo(oldContent), "should not have modified page");
+			Assert.That(newPage, Is.EqualTo(page), "should not have copied, just kept");
+		}
+
+		[Test]
+		public void MigratePageWithoutLineage_DoesNothing()
+		{
+			SetDom(@"<div class='bloom-page' id='thePage'>
+			   <div class='marginBox'>
+					<div class='bloom-imageContainer bloom-leadingElement'><img data-license='cc-by-nc-sa' data-copyright='Copyright © 2012, LASI' style='width: 608px; height: 471px; margin-left: 199px; margin-top: 0px;' src='erjwx3bl.q3c.png' alt='This picture, erjwx3bl.q3c.png, is missing or was loading too slowly.' height='471' width='608'></img></div>
+					<div aria-describedby='qtip-1' data-hasqtip='true' class='bloom-translationGroup bloom-trailingElement normal-style'>
+						<div aria-describedby='qtip-0' data-hasqtip='true' class='bloom-editable normal-style bloom-content1' contenteditable='true' lang='en'>
+							There was an old man called Bilanga who was very tall and also not yet married.
+						</div>
+					</div>
+				</div>
+			</div>
+			");
+			var book = CreateBook();
+			var dom = book.RawDom;
+			var page = (XmlElement)dom.SafeSelectNodes("//div[@id='thePage']")[0];
+			var oldContent = page.OuterXml;
+			book.BringPageUpToDate(page);
+
+			var newPage = (XmlElement)dom.SafeSelectNodes("//div[@id='thePage']")[0];
+			Assert.That(newPage.OuterXml, Is.EqualTo(oldContent), "should not have modified page");
+			Assert.That(newPage, Is.EqualTo(page), "should not have copied, just kept");
+		}
+
+		// Enhance: if there are ever cases where there are multiple image containers or multiple translation groups to migrate, test this.
+		// Enhance: if there are ever cases where it is possible not to have exactly corresponding parent elements (e.g., migrating a page with
```

Comment:
Good thought. That should produce two groups. I will use it to create another test.